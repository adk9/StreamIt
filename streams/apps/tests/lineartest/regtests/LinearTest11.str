/** Test combining pipelines when they don't rate match**/
void->void pipeline SimpleStreamItProgram {
  add FloatSource();
  add LinearPipe();
  //add LinearPipe();
  add FloatSink();
}

void->float filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

/** Pipeline of linear filters whose rates don't match. **/
float->float pipeline LinearPipe {
  add LinearFilter1(4,4);
  add ParamIdentity(2);
}

/** 
 * Simple filter that just computes a small matrix. 
 * Still keeping peek==pop.
 **/
float->float filter LinearFilter1(int numPeeks, int numPushes) {
  work peek numPeeks pop numPeeks push numPushes{
    for (int i=0; i<numPushes; i++) {
      int sum = 0;
      for (int j=0; j<numPeeks; j++) {
	sum += peek(j)*((i+1)*(j + 1) +i);
      }
      push(sum);
    }
  }
}
float->float filter ParamIdentity(int size) {
  work peek size pop size push size {
    for (int i=0; i<size; i++) {
      push(peek(i));
    }
  }
}
