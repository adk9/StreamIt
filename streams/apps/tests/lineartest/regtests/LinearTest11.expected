LinearFilter1_1.matrix
[[19.0+0.0i 14.0+0.0i 9.0+0.0i 4.0+0.0i]
 [15.0+0.0i 11.0+0.0i 7.0+0.0i 3.0+0.0i]
 [11.0+0.0i 8.0+0.0i 5.0+0.0i 2.0+0.0i]
 [7.0+0.0i 5.0+0.0i 3.0+0.0i 1.0+0.0i]]
LinearFilter1_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearFilter1_2.matrix
[[7.0+0.0i 3.0+0.0i]
 [5.0+0.0i 2.0+0.0i]
 [3.0+0.0i 1.0+0.0i]]
LinearFilter1_2.vector
[[0.0+0.0i 0.0+0.0i]]
LinearFilter2_1.matrix
[[1.0+0.0i 2.0+0.0i]
 [3.0+0.0i 4.0+0.0i]
 [5.0+0.0i 6.0+0.0i]]
LinearFilter2_1.vector
[[0.0+0.0i 0.0+0.0i]]
LinearPipe2_1.matrix
[[16.0+0.0i 26.0+0.0i 0.0+0.0i 0.0+0.0i]
 [11.0+0.0i 18.0+0.0i 0.0+0.0i 0.0+0.0i]
 [6.0+0.0i 10.0+0.0i 0.0+0.0i 0.0+0.0i]
 [35.0+0.0i 42.0+0.0i 3.0+0.0i 6.0+0.0i]
 [25.0+0.0i 30.0+0.0i 2.0+0.0i 4.0+0.0i]
 [15.0+0.0i 18.0+0.0i 1.0+0.0i 2.0+0.0i]
 [0.0+0.0i 0.0+0.0i 36.0+0.0i 46.0+0.0i]
 [0.0+0.0i 0.0+0.0i 25.0+0.0i 32.0+0.0i]
 [0.0+0.0i 0.0+0.0i 14.0+0.0i 18.0+0.0i]]
LinearPipe2_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
LinearPipe_1.matrix
[[19.0+0.0i 14.0+0.0i 9.0+0.0i 4.0+0.0i]
 [15.0+0.0i 11.0+0.0i 7.0+0.0i 3.0+0.0i]
 [11.0+0.0i 8.0+0.0i 5.0+0.0i 2.0+0.0i]
 [7.0+0.0i 5.0+0.0i 3.0+0.0i 1.0+0.0i]]
LinearPipe_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
ParamIdentity_1.matrix
[[1.0+0.0i]]
ParamIdentity_1.vector
[[0.0+0.0i]]
ParamIdentity_2.matrix
[[1.0+0.0i 0.0+0.0i]
 [0.0+0.0i 1.0+0.0i]]
ParamIdentity_2.vector
[[0.0+0.0i 0.0+0.0i]]
